week2/homework/src/index.js

```diff
@@ -1,3 +1,112 @@
 'use strict';
-
 // TODO: Write the homework code in this file
+'use strict';
+const { readFile, writeFile, appendFile } = require('fs');
+const { promisify } = require('util');
+
+const readFileWithPromise = promisify(readFile);
+const writeFileWithPromise = promisify(writeFile);
+const appendFileWithPromise = promisify(appendFile);
+
+const TODO_FILE = 'todo.json';
+
+async function main() {
+  const [, , cmd, ...args] = process.argv;
+  switch (cmd) {
+    case 'list': {
+      const data = await readFileWithPromise(TODO_FILE, `utf8`).catch(() => '[]');
+      const todos = JSON.parse(data);
+      console.info(todos);
+      break;
+    }
+
+    case 'add': {
+      const data = await readFileWithPromise(TODO_FILE, `utf8`).catch(() => '[]');
+      const todos = JSON.parse(data);
+      const newTodo = args.join(' ');
+      todos.push(newTodo);
+      await writeFileWithPromise(TODO_FILE, JSON.stringify(todos));
+      break;
+    }
+
+    case 'remove': {
+      const [, , cmd, baseIndex, ...args] = process.argv;
+      const index = parseInt(baseIndex);
+      const data = await readFileWithPromise(TODO_FILE, `utf8`).catch(() => '[]');
+      const todos = JSON.parse(data);
+
+      if ((typeof (index) === 'number' && !isNaN(index))) {
+        const todosSize = Object.keys(todos).length;
+        if (index > todosSize) {
+          console.info(`There is ${todosSize} items in the file choose a number between (1 to ${todosSize} )`);
+          return;
+        }
+        else if (index <= 0) {
+          console.info(`Please type a valid number after remove command `);
+          return;
+        }
+        else if (baseIndex == null) {
+          console.info(`Please type a number after remove command `);
+          return;
+        }
+        const items = [todos];
+        items[0].splice(index - 1, 1);
+        let newTodos = items[0];
```

Comment:
I don't understand the logic of this. Why are you storing `todos` inside of another array? You could directly do:
```js
todos.splice(index - 1, 1);
await writeFileWithPromise(TODO_FILE, JSON.stringify(todos));
console.info(todos);
```